### PR TITLE
Federation basic test

### PR DESCRIPTION
**Pull request by behcio**
## Checklist

- [ ] Make sure that tests are passing in githubaction
- [ ] Check if change should be populated to other branches (master, release-3-lts, release-4.0)
- [ ] Tests are stable

### CODE REVIEW

**Review by bojantyk — APPROVED** [review]
LGTM

File: test/specs/graphql/crud_federation_test.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    const federationExampleTestUrl = "http://localhost:4000/"
+    const api_connection = new API_connection(federationExampleTestUrl);
+    const expectedFederationResponse = { data: { me: { id: '1234' } } }
+
+    const federationTestRequestConfig = {
+        path: "query",
+        body: `{"query":"query{me{id}}","variables":{}}`
+    }    
+
+    let $supergraphTableElement;
```
Comment (konrad-sol) [inline test/specs/graphql/crud_federation_test.js:35]: just to confirm: is this table not working with our wrapper?
If it works, I think we could reuse the methods from there to click/check cells

Comment (behcio) [reply test/specs/graphql/crud_federation_test.js:35]: I'm not sure why you ask. I'm using our wrapper. Dropdown object to be precise.

File: test/specs/graphql/crud_federation_test.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+        while(isFederationExampleRunning == false && retryNumber <= federationConnectionRetries){
+            
+            console.log(`>>> Starting connection try number ${retryNumber}`);
+            
+            try {
+                response = api_connection.sendPostRequest({path: federationTestRequestConfig.path, body: federationTestRequestConfig.body});    
+                if(JSON.stringify(response.body) == JSON.stringify(expectedFederationResponse)) isFederationExampleRunning = true;
+            } catch (error) {
+                console.log('Federation example not running. Trying again. ' + error);
```
[the remark this answers is not in the document]
Comment (behcio) [reply test/specs/graphql/crud_federation_test.js:62]: Done

File: test/specs/graphql/crud_federation_test.js [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+            runFederationExample();
+        }
+
+        while(isFederationExampleRunning == false && retryNumber <= federationConnectionRetries){
+            
+            console.log(`>>> Starting connection try number ${retryNumber}`);
+            
+            try {
+                response = api_connection.sendPostRequest({path: federationTestRequestConfig.path, body: federationTestRequestConfig.body});    
+                if(JSON.stringify(response.body) == JSON.stringify(expectedFederationResponse)) isFederationExampleRunning = true;
```
[the remark this answers is not in the document]
Comment (behcio) [reply test/specs/graphql/crud_federation_test.js:60]: Done

File: test/specs/graphql/crud_federation_test.js [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+        let response;
+        login_page.open();
+        login_page.login(envDetails.userEmail, envDetails.userPassword);
+        
+        if(process.env.RUN_ENV != "CI"){
+            console.log(`RUN_ENV variable not set to CI. Running local Federation example`);
+            runFederationExample();
+        }
+
+        while(isFederationExampleRunning == false && retryNumber <= federationConnectionRetries){
```
[the remark this answers is not in the document]
Comment (behcio) [reply test/specs/graphql/crud_federation_test.js:54]: Done

File: lib/utils/utils.js [hunk 5]
```diff
@@ -8,4 +10,21 @@ const randomEmailGenerator = require('random-email');
 export const generateRandomEmail = () => {
     const randomEmail = randomEmailGenerator({ domain: 'example.com' });
     return randomEmail.replace('@', `+${moment().format('D-MMM-h_mm_ss')}@`);
+};
+
+export const runFederationExample = () => {
+    console.log('>>> Checking presence of federation start.sh file');
+    exec('ls graphql-go-tools/examples/federation | grep start.sh', (error, stdout, stderr) => {
+        if(error){
+            console.log('Error: ', error.message);
```
[the remark this answers is not in the document]
Comment (behcio) [reply lib/utils/utils.js:19]: Done

File: test/specs/graphql/crud_federation_test.js [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+import { login_page } from '../../../lib/pom/Login_page';
+import { main_page } from '../../../lib/pom/Main_page';
+import { runFederationExample } from '../../../lib/utils/utils';
+import { API_connection } from '../../../lib/utils/api_connections/API_connection';
+
+describe('CRUD simple GraphQL (proxy-only) API', () => {
+
+    let localFederationUrl;
+
+    if(process.env.RUN_ENV == "CI") localFederationUrl = "172.17.0.1";
```
[the remark this answers is not in the document]
Comment (behcio) [reply test/specs/graphql/crud_federation_test.js:12]: done

**Review by konrad-sol — APPROVED** [review]

**Pull request merged**